bin/rose-date

```diff
@@ -175,4 +173,34 @@
 #
 #     For example, for a duration `P57DT12H` - `y,m,d,h` -> `0,0,57,12`
 #-------------------------------------------------------------------------------
-exec python3 -m rose.date "$@"
+# -*- coding: utf-8 -*-
+import re
+import sys
+import os
+
+from isodatetime.main import main
+
+if __name__ == '__main__':
+    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
+
+    # Handle Legacy Rose-date -c functionality
+    if '-c' in sys.argv or '--use-task-cycle-time' in sys.argv:
+        rose_task_cycle_time = os.getenv('ROSE_TASK_CYCLE_TIME')
+        isodatetimeref = os.getenv('ISODATETIMEREF')
+        if rose_task_cycle_time:
+            os.environ['ISODATETIMEREF'] = rose_task_cycle_time
+        elif isodatetimeref:
+            pass
+        else:
+            sys.exit("[FAIL] [UNDEFINED ENVIRONMENT VARIABLE] ROSE_TASK_CYCLE_TIME")
+        if '-c' in sys.argv:
+            sys.argv.remove('-c')
+        elif '--use-task-cycle-time' in sys.argv:
+            sys.argv.remove("--use-task-cycle-time")
+        sys.argv.append('ref')
```

Comment:
Just to handle potential user badness. Perhaps:
```python
for opt in ('-c', '--use-task-cycle-time'):
    while opt in sys.argv:
        sys.argv.remove(opt)
if 'ref' not in sys.argv:
    sys.argv.append('ref')
```

Reply:
:+1: